### PR TITLE
Refactor bottom-up edit distance function to be class method

### PR DESCRIPTION
**Pull request by tianyizheng02**
### Describe your change:

Refactor bottom-up edit distance function to be part of the `EditDistance` class and renamed functions and variables for consistency

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
@ChrisO345 Your call on this one.

File: dynamic_programming/edit_distance.py [hunk 1]
```diff
@@ -19,74 +19,72 @@ class EditDistance:
...
+        >>> EditDistance().min_dist_top_down("", "")
+        0
+        """
+        self.word1 = word1
+        self.word2 = word2
+        self.dp = [[-1 for _ in range(len(word2))] for _ in range(len(word1))]
+
+        return self.__min_dist_top_down_dp(len(word1) - 1, len(word2) - 1)
+
+    def min_distance_bottom_up(self, word1: str, word2: str) -> int:
```
Comment (ChrisO345) [inline dynamic_programming/edit_distance.py:59]:
```suggestion
    def min_dist_bottom_up(self, word1: str, word2: str) -> int:
```

File: dynamic_programming/edit_distance.py [hunk 2]
```diff
@@ -99,7 +97,7 @@ def min_distance_bottom_up(word1: str, word2: str) -> int:
     S2 = input("Enter the second string: ").strip()
 
     print()
-    print(f"The minimum Edit Distance is: {solver.solve(S1, S2)}")
-    print(f"The minimum Edit Distance is: {min_distance_bottom_up(S1, S2)}")
+    print(f"The minimum edit distance is: {solver.min_dist_top_down(S1, S2)}")
+    print(f"The minimum edit distance is: {solver.min_distance_bottom_up(S1, S2)}")
```
Comment (ChrisO345) [inline dynamic_programming/edit_distance.py:101]:
```suggestion
    print(f"The minimum edit distance is: {solver.min_dist_bottom_up(S1, S2)}")
```

File: dynamic_programming/edit_distance.py [hunk 3]
```diff
@@ -19,74 +19,72 @@ class EditDistance:
...
+        """
+        self.word1 = word1
+        self.word2 = word2
+        self.dp = [[-1 for _ in range(len(word2))] for _ in range(len(word1))]
+
+        return self.__min_dist_top_down_dp(len(word1) - 1, len(word2) - 1)
+
+    def min_distance_bottom_up(self, word1: str, word2: str) -> int:
+        """
+        >>> EditDistance().min_distance_bottom_up("intention", "execution")
```
Comment (ChrisO345) [inline dynamic_programming/edit_distance.py:61]:
```suggestion
        >>> EditDistance().min_dist_bottom_up("intention", "execution")
```

File: dynamic_programming/edit_distance.py [hunk 4]
```diff
@@ -19,74 +19,72 @@ class EditDistance:
...
+        5
+        >>> EditDistance().min_distance_bottom_up("intention", "")
```
Comment (ChrisO345) [inline dynamic_programming/edit_distance.py:63]:
```suggestion
        >>> EditDistance().min_dist_bottom_up("intention", "")
```

File: dynamic_programming/edit_distance.py [hunk 5]
```diff
@@ -19,74 +19,72 @@ class EditDistance:
...
+        9
+        >>> EditDistance().min_distance_bottom_up("", "")
```
Comment (ChrisO345) [inline dynamic_programming/edit_distance.py:65]:
```suggestion
        >>> EditDistance().min_dist_bottom_up("", "")
```

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**